statistics/handle/handle.go

```diff
@@ -221,6 +221,26 @@ func (h *Handle) Update(is infoschema.InfoSchema) error {
 	return nil
 }
 
+// UpdateSessionVar updates the necessary session variables for the stats reader.
+func (h *Handle) UpdateSessionVar() error {
+	reader, err := h.getStatsReader(nil)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err1 := h.releaseStatsReader(reader)
+		if err1 != nil && err == nil {
+			err = err1
+		}
+	}()
+
+	verInString, err := reader.ctx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(variable.TiDBAnalyzeVersion)
+	ver, err := strconv.ParseInt(verInString, 10, 64)
+	reader.ctx.GetSessionVars().AnalyzeVersion = int(ver)
+	return err
+}
+
```

Comment:
How about moving this logic to `HandleAutoAnalyze`?
Then we can remove this background task.